docs/auth-flows.md

```diff
@@ -0,0 +1,142 @@
+# Authentication Flows
+
+If authentication is required, Julia uses [bearer tokens (RFC 6750)](https://datatracker.ietf.org/doc/html/rfc6750) to authenticate package server requests.
+That is, the HTTP requests set the `Authorization: Bearer $access_token` header when fetching data from the package server.
+
+The PkgAuthentication manages acquiring these tokens from package server, generally via an interactive flow.
+This document describes the details related to the authentication,
+and also acts as a specification for a few PkgAuthentication-specific conventions that _authenticated_ package servers should follow.
+
+Additional resources & references:
+
+- [Original implementation notes `JuliaLang/Pkg.jl#1538`](https://github.com/JuliaLang/Pkg.jl/pull/1538#issuecomment-564118431)
+
+## Authenticated package requests & `auth.toml` files
+
+Julia (i.e. Pkg - the package manager) stores the token information in a `auth.toml` file in the "server directory".
+For each package server host, it is generally stored as a TOML file at `~/.julia/servers/{hostname}/auth.toml`.
+
+Pkg uses the following top-level key values pairs:
+
+- `access_token` (REQUIRED): the bearer token used to authorize normal requests (string)
+- `expires_at` (OPTIONAL): an absolute expiration time (seconds from UNIX epoch; integer)
```

Comment:
We also have `expires` with the same value. Not sure why/if we need both.

Reply:
That's something custom that JuliaHub injects. Pkg only looks at `expires_at` and `expires_in`, so those are the relevant ones for this spec.

Side note: it looks like OAuth2 only uses `expires_in`, so `expires_at` is also a custom Julia invention.

Reply:
Yes, `expires_at` is again custom attribute. From Pkg specs, it says it supports only "HTTP bearer auth". The attributes it expects in `auth.toml` may look similar to OpenID payload, but we should not expect any compliance.